libcodechecker/store.py

```diff
@@ -0,0 +1,436 @@
+                       type=str,
+                       dest="dbusername",
+                       default='codechecker',
+                       required=False,
+                       help="Username to use for connection.")
+
+    pgsql.add_argument('--db-name',
+                       type=str,
+                       dest="dbname",
+                       default="codechecker",
+                       required=False,
+                       help="Name of the database to use.")
+
+    add_verbose_arguments(parser)
+
+    def __handle(args):
+        """Custom handler for 'store' so custom error messages can be
+        printed without having to capture 'parser' in main."""
+
+        def arg_match(options):
+            """Checks and selects the option string specified in 'options'
+            that are present in the invocation argv."""
+            matched_args = []
+            for option in options:
+                if any([arg if option.startswith(arg) else None
+                        for arg in sys.argv[1:]]):
+                    matched_args.append(option)
+                    continue
+
+            return matched_args
+
+        # See if there is a "PostgreSQL argument" specified in the invocation
+        # without '--postgresql' being there. There is no way to distinguish
+        # a default argument and a deliberately specified argument without
+        # inspecting sys.argv.
+        options = ['--dbaddress', '--dbport', '--dbusername', '--dbname',
+                   '--db-address', '--db-port', '--db-username', '--db-name']
+        psql_args_matching = arg_match(options)
+        if any(psql_args_matching) and \
+                'postgresql' not in args:
+            first_matching_arg = next(iter([match for match
+                                            in psql_args_matching]))
+            parser.error("argument {0}: not allowed without "
+                         "argument --postgresql".format(first_matching_arg))
+            # parser.error() terminates with return code 2.
+
+        if 'postgresql' not in args:
+            # Later called database modules need the argument to be actually
+            # present, even though the default is suppressed in the optstring.
+            setattr(args, 'postgresql', False)
+        else:
+            # If --postgresql is given, --sqlite is useless.
+            delattr(args, 'sqlite')
+
+        # If everything is fine, do call the handler for the subcommand.
+        main(args)
+
+    parser.set_defaults(func=__handle)
+
+
+def consume_plist(item):
+    f, context, metadata_dict = item
+
+    LOG.debug("Parsing input file '" + f + "'")
+
+    buildaction = build_action.BuildAction()
+    if os.path.basename(f).startswith("clangsa_"):
+        buildaction.analyzer_type = analyzer_types.CLANG_SA
+    elif os.path.basename(f).startswith("clang-tidy_"):
+        buildaction.analyzer_type = analyzer_types.CLANG_TIDY
+
+    buildaction.original_command = "IMPORTED"
```

Comment:
No, we don't. Only the `codechecker-analyzer` invocation command-line is kept... as the build-jsons could come from anywhere and in an arbitrary number.